openedx/core/djangoapps/theming/helpers_dirs.py

```diff
@@ -10,49 +10,62 @@
 from path import Path
 
 
-def get_theme_base_dirs_from_settings(theme_dirs=None):
+def get_theme_base_dirs_from_settings(theme_base_dirs=None):
     """
     Return base directories that contains all the themes.
 
     Example:
         >> get_theme_base_dirs_from_settings('/edx/app/ecommerce/ecommerce/themes')
         ['/edx/app/ecommerce/ecommerce/themes']
 
+    Args:
+        themes_base_dirs (list of str): Paths to themes base directories.
+
     Returns:
          (List of Paths): Base theme directory paths
     """
-    theme_base_dirs = []
-    if theme_dirs:
-        theme_base_dirs.extend([Path(theme_dir) for theme_dir in theme_dirs])
-    return theme_base_dirs
+    theme_base_dirs_paths = []
+    if theme_base_dirs:
+        theme_base_dirs_paths.extend([Path(theme_base_dir) for theme_base_dir in theme_base_dirs])
+    return theme_base_dirs_paths
 
 
-def get_themes_unchecked(themes_dirs, project_root=None):
+def get_themes_unchecked(themes_base_dirs, project_root=None):
     """
     Returns a list of all themes known to the system.
 
     Args:
-        themes_dirs (list): Paths to themes base directory
+        themes_base_dirs (list of str / list of Path): Paths to themes base directories.
         project_root (str): (optional) Path to project root
     Returns:
-        List of themes known to the system.
+        (list of Theme): List of themes known to the system.
     """
-    themes_base_dirs = [Path(themes_dir) for themes_dir in themes_dirs]
+    # Convert base dirs to Path objects
+    themes_base_dirs = [Path(themes_base_dir) for themes_base_dir in themes_base_dirs]
     # pick only directories and discard files in themes directory
     themes = []
-    for themes_dir in themes_base_dirs:
-        themes.extend([Theme(name, name, themes_dir, project_root) for name in get_theme_dirs(themes_dir)])
+    for themes_base_dir in themes_base_dirs:
+        themes.extend([Theme(name, name, themes_base_dir, project_root) for name in get_theme_dirs(themes_base_dir)])
 
     return themes
 
 
-def get_theme_dirs(themes_dir=None):
+def get_theme_dirs(themes_base_dir=None):
     """
-    Returns theme dirs in given dirs
+    Get all the theme dirs directly under a given base dir.
+
     Args:
-        themes_dir (Path): base dir that contains themes.
+        themes_base_dir (Path): base dir that contains themes.
+    Returns:
+        List of theme dir names (relative to the base dir) or empty list if the base themes dir does not exist or there
+        are no containing theme dirs.
     """
-    return [_dir for _dir in os.listdir(themes_dir) if is_theme_dir(themes_dir / _dir)]
+    try:
+        themes_base_dir_listing = os.listdir(themes_base_dir)
+    except FileNotFoundError:
+        themes_base_dir_listing = []
```

Comment:
This is the only functional change, and what the description refers to.  Everything else is just making the the inconsistent terminology more consistent.

Reply:
I'm trying to think about the unintended consequences of this change. If themes were not available this change will allow the app to start now. Prior to this if something had gone wrong in the automation the app just wouldn't start and we would know something was up and could halt automation. 

Perhaps it would be better for us to add a flag that could be passed in that would skip theming and allow the app to start in kind of a headless/management command only mode?

Reply:
This change will only allow the app to start IFF theming is disabled.  There were two places where app could be blocked from starting: (1) here, and (2) in the app checks.py file.  Only the latter is supposed to block anything from starting; throwing an uncaught exception from a settings file is just a simple mistake.

So, functionally, this change is a no-op.  If in production or in devstack the themes base dir was not installed, then the app will _still_ not start.  The only difference is that instead of throwing an uncaught exception the app checks.py will report `openedx.core.djangoapps.theming.E007`:

https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/theming/checks.py#L73-L80

So actually this change improves the error logging somewhat.

Reply:
Thank you for the explanation